### DISCUSSION TITLE
Performance Issues

### DISCUSSION THREAD

**Discussion by ShaungBhone** [filamentphp/filament#5997] [General]
Hello folks. These issues were found when we developed our app. In local development, we didn't notice. When we deploy to production following are give us sick.

1. Filament js file is massive.
2. Server request on every button press.

### 1. Filament js file 
- Filament js file is huge almost 1MB and renders on every page you can see rendering every page that may slow our app.
![Screenshot from 2023-03-22 18-28-58](https://user-images.githubusercontent.com/56265471/226899347-a08439a4-1946-430e-b415-efa5c12920cc.png)

### 2. Server request on every button press
- In our app, we have a quiz section for courses. That is made with a filament repeater. When I press add new repeater field every request sends each one press. That may slow our app.

[Screencast from 03-22-2023 06:40:14 PM.webm](https://user-images.githubusercontent.com/56265471/226901209-1c9e3da7-fecb-4ce1-af59-fd59af1dbfc9.webm)

- When I create new categories show modal and create. When I press the new categories button, every request sends to each one press and press the close button as well. That may slow our app.

[Screencast from 03-22-2023 06:45:31 PM.webm](https://user-images.githubusercontent.com/56265471/226902235-bfc6aecc-9657-4660-827f-557671fb75da.webm)

> Do you have any suggestions? And do we need to do something? Thank you.

**Comment by danharrin** [comment]
1) The JS file is cached by your browser. It is not re-downloaded on every page. The file is large as it contains all the JS for your form components like the rich text editor, markdown editor, file upload input, Alpine.js.

2) Server requests when you press a button are very normal in every modern web app. It's also how Filament works. We don't load content into the page unless we need to. That *would* slow things down if we did that.

Ensure that PHP opcache is enabled on your server. Ensure that you have run `icons:cache` on your server.

**Reply by ShaungBhone** [reply]
Hello, @danharrin I mean press button is like as you can see the modal is sending request every open and close. I don't think it is not normal in every modern web app. That's my opinion.

**Reply by danharrin** [reply]
Otherwise the modals content is always in the DOM. For complex modals, trust me when I say this is more efficient.

**Reply by ShaungBhone** [reply]
Thank you. I trust you that's why I used filament.

**Reply by VincvanAs** [reply]
Hi @ShaungBhone ,

Maybe a weird question, but do you maybe use   "maatwebsite/excel" in your application? Ive once had similar issues in Filament and the cause was this package. Went wrong 4/5 seconds load to 600ms after uninstall/reinstall. Still cant explain why... If your not using this specific package, i would suggest uninstall some 3th party packages to see if theres an issue there.

**Reply by ShaungBhone** [reply]
> do you maybe use   "maatwebsite/excel" in your application?

No, I didn't use any 3rd party packages. As I mentioned above screen cast, all are filament things.

**Reply by VincvanAs** [reply]
Hi Shaun,

My issue was very similar. Debugbar showed only Filament.js  and serveFilament with ordinary loading times. Even my login page was 4/5 seconds loading speed. The package was not even active within a form or table, but still able to cause performance problems

**Comment by vitalijalbu** [comment]
me too, having a large repeater content the DOM crashes
https://github.com/filamentphp/filament/discussions/19287
